js/style/style_layer.js

```diff
@@ -189,17 +190,15 @@ StyleLayer.prototype = util.inherit(Evented, {
 
         var name;
         for (name in declarations) { // apply new declarations
-            this.updatePaintTransition(name, declarations[name], options, globalOptions, animationLoop);
+            this.applyPaintDeclaration(name, declarations[name], options, globalOptions, animationLoop);
         }
         for (name in this._paintTransitions) {
             if (!(name in declarations)) // apply removed declarations
-                this.updatePaintTransition(name, null, options, globalOptions, animationLoop);
+                this.applyPaintDeclaration(name, null, options, globalOptions, animationLoop);
         }
-
-        this.updateConstantValues();
     },
 
-    updatePaintTransition: function (name, declaration, options, globalOptions, animationLoop) {
+    applyPaintDeclaration: function (name, declaration, options, globalOptions, animationLoop) {
```

Comment:
Both actually mutate `Style`. `updatePaintTransition` updates a paint transition given a paint property name and classes. `applyPaintDeclaration` does the same but given a _declaration object_ (which is derived from classes/name). 

Both `updatePaintTransition` and `updatePaintTransitions` use `applyPaintDeclaration` internally, but the latter isn't supposed to be called anywhere else, so maybe it should be underscored (`_applyPaintDeclaration`) for clarity.